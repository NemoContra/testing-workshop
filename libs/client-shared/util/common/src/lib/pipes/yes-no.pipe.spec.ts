import { createPipeFactory } from '@ngneat/spectator/jest';
import { YesNoPipe } from './yes-no.pipe';

describe('YesNoPipe', () => {
  const createPipe = createPipeFactory(YesNoPipe);

  it('should return yes for true', () => {
    const spectator = createPipe(`<span>{{ true | yesNo }}</span>`);
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should return no for false', () => {
    const spectator = createPipe(`<span>{{ false | yesNo }}</span>`);
    expect(spectator.fixture).toMatchSnapshot();
  });
});
