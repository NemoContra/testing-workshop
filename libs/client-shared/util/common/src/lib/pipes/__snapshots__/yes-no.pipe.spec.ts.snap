// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YesNoPipe should return no for false 1`] = `
<ng-component
  __ngContext__={[Function Number]}
>
  <span>
    no
  </span>
</ng-component>
`;

exports[`YesNoPipe should return yes for true 1`] = `
<ng-component
  __ngContext__="0"
>
  <span>
    yes
  </span>
</ng-component>
`;
